src/tir/transforms/lower_sparse_tir.cc

```diff
@@ -0,0 +1,306 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file lower_sparse_tir.cc
+ */
+
+#include <tvm/arith/analyzer.h>
+#include <tvm/tir/analysis.h>
+#include <tvm/tir/stmt_functor.h>
+#include <tvm/tir/transform.h>
+
+#include <utility>
+
+#include "ir_utils.h"
+
+namespace tvm {
+namespace tir {
+
+/*!
+ * \brief Check whether a given SparseBuffer contains the given axis.
+ * \brief buffer The SparseBuffer to be checked
+ * \brief axis The axis to be checked
+ * \return A boolean indicating whether the given SparseBuffer contains the given axis
+ */
+bool BufferContainsAxis(const SparseBuffer& buffer, const Axis& axis) {
+  for (int i = 0; i < static_cast<int>(buffer->axes.size()); ++i) {
+    if (buffer->axes[i].same_as(axis)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+using BufferAccessMap = Map<SparseBuffer, Array<SpIterVar>>;
+using DependencyMap =
+    std::unordered_map<SpIterVar, std::pair<SparseBuffer, int>, ObjectPtrHash, ObjectPtrEqual>;
+
+/*！
+ * \brief For each sparse-fixed or sparse-variable iterator, collect the iterators that it depends

```

Comment:
I guess what you want to ask is about dense-variable iterators.

Well, it's true that dense-variable iterators must depend on other iterators. But we don't need to know those iterators. Let me explain my reason.

Do you remember I told you that, the reason why we collect dependency information is that we want to get the dense values of iterators? Since the dense values of dense iterators are just the values of the iterators, we only need to get the dense values for sparse iterators. Therefore, we only care about the dependency information of all the sparse iterators. This is why this function only collects information for sparse iterators.